lib/msf/ui/console/command_dispatcher/jobs.rb

```diff
@@ -212,8 +212,9 @@ def cmd_jobs(*args)
               # Remove persistence by job id.
               job_list.map(&:to_s).each do |job|
                 if framework.jobs.key?(job)
-                  next unless framework.jobs[job.to_s].ctx[1] # next if no payload context in the job
-                  payload_option = framework.jobs[job.to_s].ctx[1].datastore
+                  ctx_1 = framework.jobs[job.to_s].ctx[1]
+                  next if ctx_1.nil? || !ctx_1.respond_to?(:datastore)  # next if no payload context in the job
+                  payload_option = ctx_1.datastore
```

Comment:
It turns out that the context is an untyped array, in certain scenarios it's a string:

```
lib/msf/base/simple/auxiliary.rb
73:    ctx = [mod, run_uuid, job_listener]
126:    ctx = [mod, run_uuid, job_listener]

lib/msf/base/simple/exploit.rb
194:    ctx = [mod, run_uuid, job_listener]
```

But I’m guessing this is the scenario the original code was meant to cover the scenario of the ctx being a payload:

```
lib/msf/core/exploit_driver.rb
148:      ctx = [ e, p ]
166:      ctx = [ exploit, payload ]

lib/msf/core/evasion_driver.rb
89:    ctx = [ evasion, payload ]
```